### PR TITLE
Lvtexuqj: signing journey

### PR DESCRIPTION
**Pull request by AmritSidhu**
Finishing up the user journey based on the prototype designs for uploading and replacing signing and encryption certificates. 
Created some more tests to show the signing journey working
Changed the root to now be the user-journey
Made the dates readable and in the format of the prototype
Refactored the controller to follow DRY

### CODE REVIEW

File: app/controllers/user_journey_controller.rb [hunk 1]
```diff
@@ -4,21 +4,14 @@ class UserJourneyController < ApplicationController
   include ComponentConcern
   include CertificateConcern
 
+  before_action :find_certificate
```
Comment (jakubmiarka) [inline app/controllers/user_journey_controller.rb]:
```suggestion
  before_action :find_certificate except: :index
```
I don't think the `params[:certificate_id]` is present (or used) on the index page

File: app/controllers/user_journey_controller.rb [hunk 2]
```diff
@@ -4,21 +4,14 @@ class UserJourneyController < ApplicationController
...
+
   def index
     @sp_components = SpComponent.all
```
Comment (jakubmiarka) [inline app/controllers/user_journey_controller.rb]: this should now be restricted by the team, unless GDS user

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -2,4 +2,8 @@ module ApplicationHelper
   def format_date_time(cert_date_time)
     cert_date_time.strftime("%e-%m-%Y %H:%M")
   end
+
+  def date_to_readble_long_format(cert_date_time)
```
Comment (jakubmiarka) [inline app/helpers/application_helper.rb]:
```suggestion
  def date_to_readable_long_format(cert_date_time)
```

File: spec/system/visit_user_journey_before_you_start_page_spec.rb [hunk 4]
```diff
@@ -6,35 +6,41 @@
...
-  let(:root) { PKI.new }
-  let(:x509_cert) { root.generate_encoded_cert(expires_in: 9.months) }
 
   it 'shows before you start page for msa encryption and successfully goes to next page' do
-    defaults = {
-      usage: CERTIFICATE_USAGE::ENCRYPTION,
-      value: x509_cert,
-      component: msa_component
-    }
-    Certificate.create(defaults)
```
Comment (jakubmiarka) [inline spec/system/visit_user_journey_before_you_start_page_spec.rb]: 👍

File: spec/system/visit_user_journey_check_your_certificate_page_spec.rb [hunk 5]
```diff
@@ -6,44 +6,57 @@
   before(:each) do
     login_certificate_manager_user
   end
-  
-  let(:msa_component) { create(:msa_component, encryption_certificate_id: 1) }
-  let(:sp_component) { create(:sp_component, encryption_certificate_id: 1) }
-  let(:root) { PKI.new }
-  let(:x509_cert) { root.generate_encoded_cert(expires_in: 9.months) }
 
   it 'shows check your ceritifcate page for msa encryption and successfully goes to next page' do
```
Comment (jakubmiarka) [inline spec/system/visit_user_journey_check_your_certificate_page_spec.rb]:
```suggestion
  it 'shows check your certificate page for msa encryption and successfully goes to next page' do
```

File: spec/system/visit_user_journey_check_your_certificate_page_spec.rb [hunk 6]
```diff
@@ -6,44 +6,57 @@
...
     click_button 'Continue'
     expect(current_path).to eql check_your_certificate_path(msa_component.component_type, msa_component.id, msa_component.encryption_certificate_id)
     expect(page).to have_content 'MSA'
-    expect(page).to have_content "Check you've uploaded the right certificate"
+    expect(page).to have_content 'Encryption'
     click_button 'Use this certificate'
     expect(current_path).to eql confirmation_path(msa_component.component_type, msa_component.id, msa_component.encryption_certificate_id)
   end
 
   it 'shows check your ceritifcate page for sp encryption and successfully goes to next page' do
```
Comment (jakubmiarka) [inline spec/system/visit_user_journey_check_your_certificate_page_spec.rb]:
```suggestion
  it 'shows check your certificate page for sp encryption and successfully goes to next page' do
```

File: spec/system/visit_user_journey_check_your_certificate_page_spec.rb [hunk 7]
```diff
@@ -6,44 +6,57 @@
...
     click_button 'Continue'
     expect(current_path).to eql check_your_certificate_path(sp_component.component_type, sp_component.id, sp_component.encryption_certificate_id)
     expect(page).to have_content 'SP'
-    expect(page).to have_content "Check you've uploaded the right certificate"
+    expect(page).to have_content 'Encryption'
     click_button 'Use this certificate'
     expect(current_path).to eql confirmation_path(sp_component.component_type, sp_component.id, sp_component.encryption_certificate_id)
   end
 
+  it 'shows check your ceritifcate page for msa signing and successfully goes to next page' do
```
Comment (jakubmiarka) [inline spec/system/visit_user_journey_check_your_certificate_page_spec.rb]:
```suggestion
  it 'shows check your certificate page for msa signing and successfully goes to next page' do
```

File: spec/system/visit_user_journey_check_your_certificate_page_spec.rb [hunk 8]
```diff
@@ -6,44 +6,57 @@
...
+    fill_in 'certificate_value', with: certificate.value
+    click_button 'Continue'
+    expect(current_path).to eql check_your_certificate_path(msa_component.component_type, msa_component.id, msa_component.signing_certificates[0])
+    expect(page).to have_content 'MSA'
+    expect(page).to have_content 'Signing'
+    click_button 'Use this certificate'
+    expect(current_path).to eql confirmation_path(msa_component.component_type, msa_component.id, msa_component.signing_certificates[0])
+  end
+
+  it 'shows check your ceritifcate page for sp signing and successfully goes to next page' do
```
Comment (jakubmiarka) [inline spec/system/visit_user_journey_check_your_certificate_page_spec.rb]:
```suggestion
  it 'shows check your certificate page for sp signing and successfully goes to next page' do
```

File: spec/system/visit_user_journey_confirmation_page_spec.rb [hunk 9]
```diff
@@ -6,35 +6,44 @@
...
+  end
+
+  it 'shows confirmation page for sp signing and successfully goes to next page' do
+    certificate = create(:sp_signing_certificate)
+    sp_component = certificate.component
+    visit confirmation_path(sp_component.component_type, sp_component.id, sp_component.signing_certificates[0])
+    expect(page).to have_content 'SP'
+    expect(page).to have_content 'Signing'
+    click_link 'Rotate more certificates'
+    expect(current_path).to eql root_path
```
Comment (jakubmiarka) [inline spec/system/visit_user_journey_confirmation_page_spec.rb]: should have an expectation for the signing-specific content?

**Comment by AmritSidhu** [comment]
> Looking good, a few tiny comments.
> 
> Also, feels like we're getting too heavy on the system tests. Are you able to measure the time it takes to run the tests here vs master? We might want to use the controller tests for differrent component types and certs if it's taking too long.

**Comment by AmritSidhu** [comment]
@jakubmiarka master takes 30.48 seconds to run all the tests and this branch takes 32.02 seconds

File: app/controllers/user_journey_controller.rb [hunk 10]
```diff
@@ -4,21 +4,14 @@ class UserJourneyController < ApplicationController
   include ComponentConcern
   include CertificateConcern
 
+  before_action :find_teams, :find_certificate, except: :index
```
Comment (jakubmiarka) [inline app/controllers/user_journey_controller.rb]: why are we calling the `find_teams` method before each of these actions?

Comment (AmritSidhu) [reply app/controllers/user_journey_controller.rb]: Sorry what did you mean by `this should now be restricted by the team, unless GDS user`?

**Comment by Rosa-Fox** [comment]
Looks good :) 

This differs from the prototype in that it lets us add any number of signing certificates, rather than just having 2 (primary and secondary). Guessing that functionality will be part of another task?

**Pull request merged**